pkg/obfuscator/obfuscator.go

```diff
@@ -0,0 +1,11 @@
+package obfuscator
+
+// Obfuscator is the interface which all obfuscators should implement
+type Obfuscator interface {
+	// FileName takes a filename as input and return the obfuscated name
+	FileName(string) string
+	// Contents takes a file contents as input and return the obfuscated content
+	Contents(string) string
```

Comment:
Not necessarily. The replacements for file contents could be different to what replacements are done in filenames, since file names may not permit some characters while almost everything is permitted in the file contents. At some point filename could also return an error if the replacement rule is not valid for filenames.

Reply:
then we should reflect that as part of the schema as well, otherwise it will be confusing that the same rule is used for filenames and contents

Reply:
Updated the schema.

Reply:
looks good, I think you need to update the comments to also reflect that it's line based